From d675e647e0aebf2c243ef527eb8016d1ea9a254b Mon Sep 17 00:00:00 2001
From: Trent Mick <trentm@gmail.com>
Date: Thu, 11 Apr 2019 11:15:28 -0700
Subject: [PATCH] OS-7732 more cases where vm tests are creating VMs without
 cpu_caps which can break DAPI-using tests Reviewed by: Josh Wilsdon
 <josh@wilsdon.ca> Approved by: Josh Wilsdon <josh@wilsdon.ca>

---
 src/img/test/create.test.js     | 3 ++-
 src/img/test/mk-custom-image    | 1 +
 src/img/test/try-custom-image   | 1 +
 src/vm/tests/test-blocksizes.js | 4 ++++
 src/vm/tests/test-docker.js     | 3 ++-
 5 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/src/img/test/create.test.js b/src/img/test/create.test.js
index 44b8483f..c0a0e554 100644
--- a/src/img/test/create.test.js
+++ b/src/img/test/create.test.js
@@ -20,7 +20,7 @@
  *
  * CDDL HEADER END
  *
- * Copyright (c) 2019, Joyent, Inc. All rights reserved.
+ * Copyright 2019 Joyent, Inc.
  *
  * * *
  *
@@ -261,6 +261,7 @@ test('create image from bhyve vm', function (t) {
 
     var payloadCommon = {
         brand: 'bhyve',
+        cpu_cap: 100,
         autoboot: true,
         do_not_inventory: true,
         ram: 512,
diff --git a/src/img/test/mk-custom-image b/src/img/test/mk-custom-image
index 48966630..5aa7da3d 100755
--- a/src/img/test/mk-custom-image
+++ b/src/img/test/mk-custom-image
@@ -63,6 +63,7 @@ function create_basis_zone {
 {
   "uuid": "$vm_uuid",
   "brand": "joyent",
+  "cpu_cap": 100,
   "zfs_io_priority": 30,
   "quota": 20,
   "image_uuid": "$image_uuid",
diff --git a/src/img/test/try-custom-image b/src/img/test/try-custom-image
index 39b8ffa2..697e8d71 100755
--- a/src/img/test/try-custom-image
+++ b/src/img/test/try-custom-image
@@ -71,6 +71,7 @@ vmadm create <<EOM
 {
   "uuid": "$vm_uuid",
   "brand": "joyent",
+  "cpu_cap": 100,
   "zfs_io_priority": 30,
   "quota": 20,
   "image_uuid": "$image_uuid",
diff --git a/src/vm/tests/test-blocksizes.js b/src/vm/tests/test-blocksizes.js
index 9f5a8191..d84cc030 100644
--- a/src/vm/tests/test-blocksizes.js
+++ b/src/vm/tests/test-blocksizes.js
@@ -246,6 +246,7 @@ test('create zone with data_recsize 64k', function(t) {
     var payload = {
         alias: 'test-blocksizes-' + process.pid,
         brand: 'joyent-minimal',
+        cpu_cap: 100,
         autoboot: false,
         image_uuid: image_uuid,
         do_not_inventory: true,
@@ -459,6 +460,7 @@ test('create zone with compression', function(t) {
     var payload = {
         alias: 'test-blocksizes-' + process.pid,
         brand: 'joyent-minimal',
+        cpu_cap: 100,
         autoboot: false,
         image_uuid: image_uuid,
         do_not_inventory: true,
@@ -593,6 +595,7 @@ test('create KVM with block_size 64k', function(t) {
     var payload = {
         alias: 'test-blocksizes-' + process.pid,
         brand: 'kvm',
+        cpu_cap: 100,
         autoboot: false,
         do_not_inventory: true,
         ram: 128,
@@ -736,6 +739,7 @@ test('create KVM with compression', function(t) {
     var payload = {
         alias: 'test-blocksizes-' + process.pid,
         brand: 'kvm',
+        cpu_cap: 100,
         autoboot: false,
         do_not_inventory: true,
         ram: 128,
diff --git a/src/vm/tests/test-docker.js b/src/vm/tests/test-docker.js
index 58a30567..a725d17b 100644
--- a/src/vm/tests/test-docker.js
+++ b/src/vm/tests/test-docker.js
@@ -20,7 +20,7 @@
  *
  * CDDL HEADER END
  *
- * Copyright (c) 2018, Joyent, Inc.
+ * Copyright 2019 Joyent, Inc.
  *
  */
 
@@ -51,6 +51,7 @@ var common_payload = {
     alias: 'test-docker-' + process.pid,
     autoboot: false,
     brand: 'joyent-minimal',
+    cpu_cap: 100,
     do_not_inventory: true,
     max_locked_memory: 512,
     max_physical_memory: 512,
-- 
2.21.0

